app/controllers/candidate_interface/sign_up_controller.rb

```diff
@@ -9,10 +9,10 @@ def create
 
       if @candidate.persisted?
         MagicLinkSignIn.call(candidate: @candidate)
-        render 'candidate_interface/sign_in/show'
+        render 'candidate_interface/shared/check_your_email'
       elsif @candidate.save
         MagicLinkSignUp.call(candidate: @candidate)
-        render :show
+        render 'candidate_interface/shared/check_your_email'
       else
         render :new
       end
```

Comment:
As discussed this is good enough, just have a look at this implementation (not need to use it! I've just had this idea that I wanted to share with you :) )

```Ruby
  class SignUpController < CandidateInterfaceController
    def new
      @candidate = Candidate.new
    end

    def create
      @candidate = Candidate.find_or_initialize_by(candidate_params)

      if @candidate.valid?
        magic_link_email.call(candidate: @candidate)
        render 'candidate_interface/shared/check_your_email'
      else
        render :new
      end
    end

    def show; end

  private

    def candidate_params
      params.require(:candidate).permit(:email_address)
    end

    def magic_link_email
      @candidate.new_record? ? MagicLinkSignUp : MagicLinkSignIn
    end
  end
```

Reply:
Thank you for your suggestion! I can see why you would want to refactor this as we have the `render 'candidate_interface/shared/check_your_email'` duplicated. However I think the current implementation is fine for now, I kind of like how I can read through it in one go i.e. don't need read through and then refer to a private method and think about what the ternary does and then go back to the create method.